cookbook-front-end/src/Components/Recipes.tsx

```diff
@@ -35,35 +36,33 @@ query Recipes{
 `;
 
 const Recipes = () => {
+  let { path, url } = useRouteMatch();
+
   const classes = useStyles();
   const { loading, error, data } = useQuery(GET_RECIPES);
   const client = useApolloClient();
   if(loading) return <CircularProgress/>;
   if(error) return <span>oh no, an error occured</span>;
+
+  console.log("In recipes");
+  console.log(path);
+  console.log(url);
+
   return <>
         <div className={classes.recipesRoot}>
-            {data.recipes.map((r: any) => {
-                return <Card className = {classes.cardRoot} key={r.recipeId}>
-                    <CardHeader title={r.name} className={classes.cardTitle}/>
-                    <CardActions>
-                        <Button 
-                            size="small"
-                            onClick={()=> client.writeData({
-                                data: {
-                                    currentPanel: Panels.recipeDetail,
-                                    recipeDetailId: r.recipeId
-                                },
-                            })}>
-                            View Recipe
-                        </Button>
-                    </CardActions>
-                    <CardMedia
-                        className={classes.media}
-                        image={r.imageUrl}
-                        title={r.name}
-                    />
-                </Card>
-            })}
+          {data.recipes.map((r: any) => {
+              return <Card className = {classes.cardRoot} key={r.recipeId}>
+                  <CardHeader title={r.name} className={classes.cardTitle}/>
+                  <CardActions>
+                      <Link to={`recipes/${r.recipeId}`}>View recipe</Link>
```

Comment:
We don't have to have a `<RecipeDetail>` component here, because in `App.tsx` we've defined what to do when we see a route like this.

Reply:
oh very nice - yeah the syntax is so natural here it just tells you exactly what it's doing. Very cool stuff.